### DISCUSSION TITLE
Open Devtools in Production with react-query v4

### DISCUSSION THREAD

**Discussion by morenyang** [TanStack/query#3929] [Q&A] [answered]
For some reason, we want to enable the devtools in production for developers to debug the app. 
In react-query v3, we can import the devtools like:
```js
import { ReactQueryDevtools } from 'react-query/devtools/development';
```

but in v4, it doesn't seem to provide an entry point for the same purpose. 
I have tried the solution that mention in https://github.com/TanStack/query/discussions/3296#discussioncomment-3242588
```js
import { ReactQueryDevtools } from '@tanstack/react-query-devtools/src/devtools';
// and
import { ReactQueryDevtools } from '@tanstack/react-query-devtools/build/cjs/packages/react-query-devtools/src/devtools';
```
but they didn't work and with an error from webpack:
```
Module not found: Error: Package path ./src/devtools is not exported from package <my_project_path>\node_modules\@tanstack\react-query-devtools (see exports
 field in <my_project_path>\node_modules\@tanstack\react-query-devtools\package.json)
```

Is there any good way to open the devtools in production with v4? 
Thanks.

My environment:
```
react-scripts@5.0.1
typescript@4.6.2
react@18.2.0
@tanstack/react-query@4.0.10
```

**Comment by TkDodo** [comment, marked as answer]
We'll re-add that feature once we have merged the ESM bundling branch.

**Reply by morenyang** [reply]
Sounds great, thanks!

**Reply by Happybajwa** [reply]
I am still getting the issue. I am trying to make react-devtool work in production for testing purposes. const ProdDevTools = () => {
  const queryClient = new QueryClient()
  const [showDevtools, setShowDevtools] = React.useState(false)

  const ReactQueryDevtoolsProduction = React.lazy(() =>
    import('@tanstack/react-query-devtools/src/production').then(
      (d) => ({
        default: d.ReactQueryDevtools,
      }),
    ),
  )
  React.useEffect(() => {
    // @ts-ignore
    window.toggleDevtools = () => setShowDevtools((old) => !old)
  }, [])
  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter basename="/delivereasy">
        <App />
      </BrowserRouter>
      <ReactQueryDevtools initialIsOpen />
      {showDevtools && (
        <React.Suspense fallback={null}>
          <ReactQueryDevtoolsProduction />
        </React.Suspense>
      )}
    </QueryClientProvider>
  )
} and error i am getting is Module not found: Error: Package path ./src/production is not exported
![Screenshot 2023-12-15 101240](https://github.com/TanStack/query/assets/89829261/ad481aa4-db5a-461a-9aac-38e71df8f677)

**Reply by TkDodo** [reply]
looks like your bundler doesn't understand ESM, so with v5, you'd need this:

```
import('@tanstack/react-query-devtools/build/modern/production.js')
```

as described in the docs: https://tanstack.com/query/v5/docs/react/devtools#devtools-in-production

**Reply by kab1012** [reply]
@TkDodo It still doesn't start the devtools in v5 and above

"@tanstack/react-query": "^5.14.2",
"@tanstack/react-query-devtools": "^5.14.2",

**Reply by TkDodo** [reply]
please file an issue with a minimal reproduction

**Reply by kab1012** [reply]
@TkDodo i was able to run the devtools in vercel deployment .

I still have a question:
in the prod build, even though i have a staleTime and cachetime of 5 mins and 2 mins respectively, the api is always "fetching" but surprisingly there is an api request only after 2 mins. Any thoughts?

**Reply by TkDodo** [reply]
no? show a reproduction please :)

**Comment by kab1012** [comment]
@TkDodo It still doesn't start the devtools in v5 and above

"@tanstack/react-query": "^5.14.2",
"@tanstack/react-query-devtools": "^5.14.2",
